src/main.rs

```diff
@@ -268,6 +268,11 @@ impl RandomDataSource {
 }
 
 impl DataSourceMut for RandomDataSource {
+    fn fetch_description(&self) -> DataSourceDescription {
+        DataSourceDescription {
+            source_locator: vec![String::from("Random Data Source")],
```

Comment:
`.to_string()`?

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/0e7dc2a5f0870a89041fa997012d777d3df6bfaa still trying to get used to, the Rust Book seemed to prefer `String::from` and calling methods on literals is just always going to look odd to me :)